streaming/src/main/scala/org/apache/spark/streaming/scheduler/BatchInfo.scala

```diff
@@ -68,3 +83,20 @@ case class BatchInfo(
    */
   def numRecords: Long = streamIdToInputInfo.values.map(_.numRecords).sum
 }
+
+@DeveloperApi
+object BatchInfo {
+
+  /**
+   * Create `BatchInfo`. This is for binary compatibility.
+   */
+  def apply(
```

Comment:
This is also a new public method. Lets now add any public method since I dont want to expose new public methods without thinking. Also, technically they shouldnt be creating the objects of this class anyways. Also, it dont see a real need to expose a companion object.